glossary: Add deposed (resource) definition (#784)
hashicorp/terraform-website: content/source/docs/glossary.html.md

- [Terraform docs: Data Sources](/docs/configuration/data-sources.html)

## Deposed

[deposed]: glossary.html#deposed
[deposed resource]: glossary.html#deposed
[deposed resources]: glossary.html#deposed

> This status tracks a [resource][] that was marked for deletion from a previous replacement and is no longer used in references, but still remains in the Terraform [state][] and infrastructure due an error from that previous [apply][]. Terraform expected to replace the existing resource by creating a new resource then destroying the existing resource, but the error occurred before the destruction. Terraform will destroy the existing (now previous) resource next apply. This only can occur in resource [configurations][] that include the `lifecycle` configuration block `create_before_destroy` [argument][] set to `true`.

Reviewer: ```suggestion
This status tracks a [resource][] that was marked for deletion, but still remains in the Terraform [state][] and infrastructure due an error from a previous [apply][]. Terraform expected to replace the existing resource by creating a new resource, then destroying the existing resource, but an error occurred in the apply before the destruction. Existing references to the resource refer to the new resource. Terraform will destroy the `deposed` resource on the next apply. This only can occur in resource [configurations][] that have the `lifecycle` configuration block `create_before_destroy` [argument][] set to `true`.
```

Author: Great suggestion, applying! 😄
